### PR TITLE
Fix "listener not found" warning when setting `maxBounds`

### LINKED ISSUE #7951: "Listener not found" warning when passing maxBounds option to map
**Steps to reproduce**

1. Open the "debug/map/max-bounds.html" on current `main` branch
2. Look in the browser's JS console

**Expected behavior**

No error or warning messages should be present in the JS console.

**Current behavior**

A spurious "Listener not found" warning is logged by Leaflet. The warning is "harmless" in that it does not appear to break any map behavior, but it is noise for developers using Leaflet and a potential troubleshooting distraction to anyone else who might be peeking into the JS console on a project.

Screenshot of demo map with the warning visible underneath via open developer tools pane

**Environment**
- Leaflet version: testing at commit `409c17c` (which is unrelated to this issue, just happens to be at tip)
- Browser (with version): presumedly any, appears to be a logic issue
- OS/Platform (with version): presumedly any

**Additional context**

The warning and/or the conditions leading to it seem to have been introduced after the last official (1.7.1) release but I did not bisect to determine the exact commit.  It appears that the first call to `setMaxBounds` tries to unregister a `this._panInsideMaxBounds` listener before it has ever registered one.

**Minimal example reproducing the issue**

The "debug/map/max-bounds.html" example in this repo is sufficient.

### PR DESCRIPTION
**Pull request by mourner**
Fixes #7951. A simpler alternative to #8161.

### CODE REVIEW

File: src/map/Map.js [hunk 1]
```diff
@@ -446,11 +446,14 @@ export var Map = Evented.extend({
...
-			return this.off('moveend', this._panInsideMaxBounds);
-		} else if (this.options.maxBounds) {
+		if (this._maxBoundsSet) {
+			this._maxBoundsSet = false;
 			this.off('moveend', this._panInsideMaxBounds);
+
+			if (!bounds.isValid()) {
+				this.options.maxBounds = null;
+				return this;
+			}
```
Comment (Falke-Design) [inline src/map/Map.js:456]: If the variable `bounds` is passed invalid, you don't return (like it was before) and apply it to `maxBounds` in line 459.

**Review by Falke-Design — CHANGES_REQUESTED** [review]
Is a valid approche but it does **not** close #8161 it makes it only for this usecase obsolete.
The ability to check for an event handler with a certain function is missing generally and is provided with #8161.

File: src/map/Map.js (hunk 1 above)
Comment (mourner) [reply src/map/Map.js:456]: @Falke-Design what does the `return this` line do then?

**Comment by mourner** [comment]
> Is a valid approche but it does **not** close #8161 it makes it only for this usecase obsolete.

Yeah, but I don't think such a feature is necessary given that it complicates the API, unless more use cases arise (and they haven't in 11 years as far as I can tell). Leaflet's philosophy is minimalism and doing away with less.

File: src/map/Map.js (hunk 1 above)
Comment (Falke-Design) [reply src/map/Map.js:456]: If the bounds are invalid but `_maxBoundsSet` is not set, you don't come to the part `if (!bounds.isValid()) {`

Comment (mourner) [reply src/map/Map.js:456]: Oh, I see now, this is the case where max bounds wasn't set before.

**Comment by Falke-Design** [comment]
I had this usecase many times ... 

> complicates the API

I don't really see that it makes it more complex, also I think that if we provide adding (`on`) / removing (`off`), we should offer a getting (`listens`).
But let us move this disscussion to #8161

**Review by Falke-Design — APPROVED** [review]

**Comment by mourner** [comment]
@Falke-Design you're right — now that I think about it more, it makes sense to make this consistent with other methods, and you're more qualified than me with regards to possible use cases because I wasn't very active in Leaflet development for so long. Let's land that one too then.

**Comment by dmytro-shchurov** [comment]
Why do I see this warning in `"version": "1.8.0",` updated today? I mean different warning about missed listeners (moveend including)

**Comment by IvanSanchez** [comment]
> Why do I see this warning in `"version": "1.8.0",` updated today? I mean many different warnings about missed listeners (moveend including)

Because 1.8.0 was released on the 18th of April, and #8168 was merged into `main` on the 20th of April.

I mean, man, look at the dates.

**Pull request merged**